v2/spiffeid/trustdomain.go

```diff
@@ -2,63 +2,93 @@ package spiffeid
 
 import (
 	"net/url"
+	"strings"
 )
 
 // TrustDomain is the name of a SPIFFE trust domain (e.g. example.org).
 type TrustDomain struct {
+	name string
 }
 
 // TrustDomainFromString returns a new TrustDomain from a string. The string
 // can either be the host part of a URI authority component (e.g. example.org),
 // or a valid SPIFFE ID URI (e.g. spiffe://example.org), otherwise an error is
 // returned.  The trust domain is normalized to lower case.
 func TrustDomainFromString(s string) (TrustDomain, error) {
-	panic("not implemented")
+	if !strings.HasPrefix(s, "spiffe://") {
```

Comment:
This heuristic won't do the right thing if passed `http://foo/bar` (in which case the function will succeed with a trust domain of "http") nor does it handle mixed casing in the scheme (e.g. `spiFFe`). What if we only added the `"spiffe://"` prefix if the string doesn't have `"://"` in it?
```suggestion
	if strings.Index(s, "://") < 0 {
```